presto-parser/src/main/java/com/facebook/presto/sql/tree/RollbackToSavepoint.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public final class RollbackToSavepoint
```

Comment:
Another example is `ALTER TABLE`. It produces multiple nodes: `RenameTable`, `AddColumn`, `RenameColumn`

The grammar just happens to start them all with the same prefix:
```
ALTER TABLE name RENAME TO new_name
ALTER TABLE name ADD COLUMN column_name data_type
ALTER TABLE name RENAME COLUMN column_name TO new_column_name
```

Yet we could easily imagine them as being different syntax:
```
RENAME TABLE name TO new_name
ADD NEW COLUMN column_name data_type TO name
RENAME COLUMN column_name IN name TO new_column_name
```
But the exact syntax doesn't change the fundamental nature of the distinct operations. The AST represents *abstract* syntax.